### DISCUSSION TITLE
SQL snippets not available in self hosted instance?

### DISCUSSION THREAD

**Discussion by dominikjannis** [supabase/supabase#12388] [Questions] [answered]
Hi everyone,
in self hosted instances i have no possibility to save the SQL queries created via the SQL Editor. Did i misconfigure anything or is this expected behaviour? Would love to use this feature in self hosted as well, as it makes working with the db way easier.
Thanks!

**Comment by ruggi99** [comment, marked as answer]
Hi @dominikjannis ,
unfortunately this feature is not provided in Supabase self-hosted version.
It's a feature, among others, provided only in cloud hosted because it's an extension of the Supabase ecosystem.
This is also designed like this to not oversize Database with data as it is limited in Free Tier for example.
So they have a separate db to store all sql queries from all users (at least I think) and the other "external" features.
Porting this feature into self-hosting isn't quite impossible but requires Supabase team to agree on this.
For examples they (or someone) could develop a Postgres extension to be installed on self-hosted that provides migrations for storing sql queries.
Or they could insert sql migrations in the initial migrations for self-hosted version and link self-hosted api to that schema.

Tagging @kiwicopple to have some insights if this is possible in the long term.

**Reply by MildTomato** [reply]
we are tracking a feature request for this [here](https://github.com/orgs/supabase/discussions/13774)
